ui/src/geometry/geometry-api.ts

```diff
@@ -273,12 +275,22 @@ export async function getTrackLayoutPlans(
     ).then((plans) => plans.filter(filterNotEmpty));
 }
 
-export async function fetchProjects(): Promise<Project[]> {
-    return await getThrowError<Project[]>(`${GEOMETRY_URI}/projects`);
+export async function getProjects(): Promise<Project[]> {
+    const changeTime = getChangeTimes().project;
+
+    return projectCache.get(changeTime, undefined, () =>
+        getThrowError<Project[]>(`${GEOMETRY_URI}/projects`),
+    );
 }
 
 export async function getProject(id: ProjectId): Promise<Project> {
-    return await getThrowError<Project>(`${GEOMETRY_URI}/projects/${id}`);
+    return getProjects().then((projects) => {
+        const project = projects.find((project) => project.id === id);
+        if (project === undefined) {
```

Comment:
Tässä nimenomaisessa keississä ei niin väliä, mutta meillä on monessa paikassa sekaisin `null`:eja ja `undefined`:eja, sillä bäkkäri yleensä palauttaa puuttuviksi kenttien arvoiksi ym. `null`:eja (näiden yhtenäistämisestäkin on ollut puhetta, ei vaan olla toteutettu vielä.) Ollaan siten yleensä tehty nämä vertailut `!<arvo>` kun se tajuaa ne molemmat. Tässä toki ei niin väliä, kun `.find()` palauttaa aina `undefined`:in jos ei löydy

Reply:
Muutin tämän tarkistamaan falsy-arvoja eksplisiittisen `undefined`-arvon sijaan. 

Suosin itsekin tätä `if (!...)` tyyliä, mutta tällaisia if-checkkejä näkyy olevan suht sama määrä kummallakin tavalla tässä koodikannassa (greppailin siis `if(!` ja `== undefined`)